### DISCUSSION TITLE
Graphical Collaborative Tooling for Vocabulary Mapping

### DISCUSSION THREAD

**Discussion by haydenbspence** [OHDSI/Vocabulary-v5.0#848] [Ideas]
This is a preliminary analysis of tooling that may assist in the collaborative process of mapping vocabularies. They are presented in no specific order. Each has been evaluated on the following basic criteria:

- Has a graphical user interface and does not require command-line usage
- Provides real-time collaboration options
- Supports the primary task of mapping two or more vocabularies over creating or editing ontologies
- Is one or more of the following (open source, free, low cost)
- Has a moderate to easy learning curve

| Title | Description | Pros | Cons |
|-------|-------------|------|------|
| Gra.fo | A visual, collaborative, real-time ontology editor. | Strong real-time collaboration, graphical user interface, ability to handle complex relationships. | Not specifically designed for medical ontologies, some features might require a subscription. |
| Arches | An open-source platform designed for managing and disseminating cultural heritage data. | Open-source, robust mapping features, adaptable to various data types. | Originally designed for historical and archaeological data, requires some setup and configuration. |
| WebProtégé | A free, open-source platform for collaborative ontology editing developed by Stanford University. | Supports collaborative ontology editing, flexible user interface for building and managing ontologies. | Learning curve can be steep for non-technical users, not specifically designed for vocabulary mapping. |
| TopBraid EDG | A tool to manage metadata about various types of data, and offers mapping and integration features for handling various vocabularies. | Supports a broad range of data governance, robust mapping and integration features. | Not open-source, full features require paid version. |
| BioPortal | Web-based platform for sharing, browsing, and using ontologies in various formats. | Great for managing biomedical ontologies, wide range of supported formats. | Not designed for real-time collaboration, interface can be complex for new users. |
| OntoGraph | A web-based tool for visualizing, navigating, and editing RDF graphs. | Good for working with RDF graphs, visual navigation features. | Not explicitly designed for collaboration, learning curve for those unfamiliar with RDF. |
| Protege | An open-source platform for constructing domain models and knowledge-based applications with ontologies. | Powerful ontology editor, strong community support. | Steep learning curve, not designed specifically for real-time collaboration. |
| CmapTools | Tool designed for creating concept maps to represent semantic relationships. | User-friendly interface, suitable for creating visual representations. | Not specifically designed for ontology or vocabulary mapping. |
| Apache Jena | A Java framework for building Semantic Web and Linked Data applications. | Robust RDF, RDFS, OWL, and SPARQL support, includes rule-based inference engine. | Requires programming knowledge, not specifically designed for collaboration. |
| yEd Graph Editor | A tool for creating diagrams, including those necessary for vocabulary mapping. | High flexibility for different types of diagrams, user-friendly interface. | Not explicitly designed for ontology or vocabulary mapping, learning curve for complex diagrams. |
| MindMup | A user-friendly mind mapping tool that allows creating and sharing interactive mind maps. | Easy to use, supports collaboration, exports to various formats including JSON. | Not specifically designed for vocabulary mapping, advanced features may require paid version. |
| XMind | A full-featured mind mapping and brainstorming tool. | Broad range of diagram types, exports to various formats, including JSON. | Not specifically designed for vocabulary mapping, full features require paid version. |
| Freeplane | An open-source software for creating mind maps. | Open-source, supports creating complex connections and relationships. | Not specifically designed for ontology or vocabulary mapping. |
| Draw.io (diagrams.net) | A free online diagram software for making flowcharts, process diagrams, org charts, UML, ER, and network diagrams. | Wide variety of diagram types, intuitive user interface. | Not specifically designed for vocabulary mapping, advanced features might require some learning. |

This list has been evaluated with the same basic criteria aside from the graphical user interface. This is for those proficient in R or Python. It's also assumed that git would satisfy the collaboration criteria as these tools are all code.

| Title | Description | Pros | Cons |
|-------|-------------|------|------|
| SKOS (Python) | Simple Knowledge Organization System provides a model for expressing the basic structure and content of concept schemes. | Widely used in the semantic web, good interoperability. | Requires understanding of RDF and semantic web concepts. |
| Owlready2 (Python) | A module for ontology-oriented programming in Python, it can load OWL 2.0 and RDFs ontologies and manipulate them transparently in Python. | Simplifies working with ontologies in Python, supports reasoning via HermiT. | Limited to Python, might have a learning curve for some users. |
| RDFLib (Python) | A Python library for working with RDF, a fundamental format for semantic web data. | Simplifies parsing, serializing, and manipulating RDF. | Understanding of RDF and semantic web concepts is necessary. |
| SPARQLWrapper (Python) | A simple Python wrapper around a SPARQL service to remotely execute your queries, useful for querying semantic web data. | Simplifies execution of SPARQL queries from Python. | Requires understanding of SPARQL and semantic web concepts. |
| rdflib.js (JavaScript) | A JavaScript library for parsing, serializing, and manipulating RDF data, useful if you're also comfortable with JavaScript. | Allows RDF manipulation in JavaScript. | Requires understanding of RDF and semantic web concepts, and JavaScript proficiency. |
| tidygraph (R) | An R package for graph manipulation that allows you to work with graph data structures. | Simplifies graph manipulation in R, integrates with the 'tidyverse'. | Requires proficiency in R, not specific to ontologies. |
| igraph (R) | A collection of network analysis tools with the emphasis on efficiency, portability, and ease of use. | Powerful network analysis features, supports R and Python. | Can have a steep learning curve, not specific to ontologies. |

**Comment by DrUrwin** [comment]
We've developed a tool at the University of Nottingham called Carrot Mapper, it is used within the UK and internationally to map source data to the OMOP CDM, please feel free to have a look here:

https://carrot4omop.ac.uk/
https://github.com/Health-Informatics-UoN/carrot-docs

Carrot Mapper is:
- Free to use
- Open Source
- Open to collaboration for further development
